### PR TITLE
add cname record support

### PR DESCRIPTION
**Pull request by hi-glenn**

### CODE REVIEW

**Comment by mwarning** [comment]
Imho, cname support does not really adds value to this example server...

**Comment by hi-glenn** [comment]
I want to add all the record types commonly used.

It is convenient for others to learn.

Would you accept me?

Maybe the more, the better. 😃

**Comment by mwarning** [comment]
This project is meant as an example to introduce the DNS basics. I would like to keep the current feature set. How about you create a new project that implements all the features?

File: main.c [hunk 1]
```diff
@@ -119,7 +119,7 @@ union ResourceData {
     char *name;
   } name_server_record;
   struct {
-    char name;
+    char* name;
```
Comment (mwarning) [inline main.c:122]: I missed that one :)

Comment (glennWang) [reply main.c]: OK, I will create a new project that implements all the features.

Thank you very much.

**Comment by hi-glenn** [comment]
> This project is meant as an example to introduce the DNS basics. I would like to keep the current feature set. How about you create a new project that implements all the features?



OK, I will create a new project that implements all the features.

Thank you very much.

**Comment by mwarning** [comment]
Thanks for your contribution so far!

**Comment by mwarning** [comment]
btw. it is possible to unfork a forked repository. this gives your project more visibility.

**Comment by hi-glenn** [comment]
Thanks  for your advice.

**Pull request closed without merging**